scene/2d/tile_map.cpp

```diff
@@ -329,6 +331,7 @@ void TileMap::_update_dirty_quadrants() {
 				xform.set_origin(q.pos);
 				vs->canvas_item_set_transform(canvas_item, xform);
 				vs->canvas_item_set_light_mask(canvas_item, get_light_mask());
+				vs->canvas_item_set_z_index(canvas_item, z_index);
```

Comment:
Is there a way to do this without creating a new canvas item for each z index? This is probably really inefficient.

Reply:
The problem is that in many cases (IE, Isometric) drawing order is more important than reducing state changes. For materials it makes sense, but since this completely skips drawing order as the z index changes, maybe tiles could be somehow grouped by z_index when drawn..

Reply:
You mean draw the lowest z index tiles first, and up from that? I could try I guess.